docs/manual/developer_guide.adoc

```diff
@@ -630,6 +630,59 @@ Checks are connected to rules by the `oval` rule element.
 Remediations (i.e. fixes) are assigned to rules based on their basename.
 Therefore, the rule `sshd_print_last_log` has a `bash` fix associated as there is a `bash` script `shared/fixes/bash/sshd_print_last_log.sh`. As there is an Ansible playbook `shared/fixes/ansible/sshd_print_last_log.yml`, the rule has also an Ansible fix associated.
 
+==== Rule Directories
+
+To improve the organizational structure of `scap-security-guide`, we have
+introduced a new type of rule structure, a rule directory. Under the previous
+structure, a rule and its associated checks and fixes would look like this:
+
+----
+linux_os/guide/system/group/rule_id.rule
+ol7/fixes/bash/rule_id.sh
+rhel7/checks/oval/rule_id.xml
+shared/checks/oval/rule_id.xml
+shared/fixes/bash/rule_id.sh
+----
+
+Because each of the relevant files were in a different location, pointing a
+contributor to the right place involved looking in a significant number of
+directories. The rule directory simplifies this structure immensely: in a
+guide, a rule file becomes a directory with the following structure:
+
+----
+linux_os/guide/system/group/rule_id/rule.yml
+linux_os/guide/system/group/rule_id/bash/ol7.sh
+linux_os/guide/system/group/rule_id/bash/shared.sh
+linux_os/guide/system/group/rule_id/oval/rhel7.xml
+linux_os/guide/system/group/rule_id/oval/shared.xml
+----
```

Comment:
I don't think that we need lines 635-660. This would be better served in an announcement or release email and not in a "how do we develop" guide.

However, what is missing that is very important is how to develop a new rule in this new structure. What can/can't I put in this new structure? Are there any requirements? Anything I need to be aware of or know what/how to do to develop?

Reply:
I think the latter half (example layout) is important for a "how do we develop" guide, though I will remove the comparison to the original layout.


I also feel like a prereq to this PR is documenting what a `rule_id.rule` (alt: `rule.yml`) is -- nowhere does it list how to write YAML rules! :D

Reply:
@redhatrises How does the new language work?

Reply:
Okay, removed.